zipp/__init__.py

```diff
@@ -243,6 +244,14 @@ def __init__(self, root, at=""):
         self.root = FastLookup.make(root)
         self.at = at
 
+    def __eq__(self, other):
+        if self.__class__ is not other.__class__:
+            return NotImplemented
```

Comment:
I don't believe this behavior is tested.

Reply:
Coverage confirmed it was untested. In 83252516da213494307bc4fbd6191b49b255d8a0, I added a test to cover it.